File: ruby/spec/euler_spec.rb
require 'spec_helper'

describe 'Euler Problems' do
  describe SumOfMultiples do
    context 'sum of multiples' do
      subject { described_class.new }
      let(:test) { 23 }
      let(:answer) { 233168 }

      it { expect(subject.sum_multiples(10)).to eq(test) }

      it { expect(subject.sum_multiples(1000)).to eq(answer) }
    end
  end

  describe SumEvenFib do
    context 'sum even fibonacci numbers' do
      subject { described_class.new }
      let(:test) { 44 }
      let(:answer) { 4613732 }

      it { expect(subject.sum_even_fib(89)).to eq(test) }

      it { expect(subject.sum_even_fib(4000000)).to eq(answer) }
    end
  end

  describe PrimeFactor do
    context 'find largest prime factor' do
      subject { described_class.new }
      let(:test) { 29 }
      let(:answer) { 6857 }

      it { expect(subject.largest_prime(13195)).to eq(test) }

      it { expect(subject.largest_prime(600851475143)).to eq(answer) }
    end
  end
end

File: javascript/spec/javascripts/euler_problems/EulerSpec.js
describe("SumofMultiples", function() {
  var sumOfM;

  it("should sum multiples up to 10", function() {
    sumOfM = new SumOfMultiples();
    var result = sumOfM.sum(10);

    expect(result).toEqual(23);
  });

  it("should sum multiples up to 1000", function() {
    sumOfM = new SumOfMultiples();
    var result = sumOfM.sum(1000);

    expect(result).toEqual(233168);
  });
});
